.github/workflows/push-to-testnet.yml

```diff
@@ -24,62 +24,10 @@ jobs:
           echo "FIRMWARE_SHORT_HASH=$FIRMWARE_SHORT_HASH" >> $GITHUB_ENV
       - name: Random sleep (balena API workaround)
         run:  sleep $[ ( $RANDOM % 360 ) + 1 ]s
-      - name: Generate Docker Compose for RockPi
-        if: matrix.sbc == 'rockpi'
+      - name: Generate Docker Compose
         run: |
-          python gen_docker_compose.py rockpi -o device-compose-files/docker-compose-rockpi.yml
-          cp device-compose-files/docker-compose-rockpi.yml docker-compose.yml
-          ROCKPI="-rockpi"
-          echo "ROCKPI=$ROCKPI" >> $GITHUB_ENV
-      - name: Generate Docker Compose for RaspberryPi & RAK & Other RasPi based vendors
-        if: matrix.sbc == 'raspi' || matrix.sbc == 'rak' || matrix.sbc == 'og' || matrix.sbc == 'sensecap' || matrix.sbc == 'finestra' || matrix.sbc == 'controllino' || matrix.sbc == 'cotx' || matrix.sbc == 'pantherx1'
-        run: |
-          python gen_docker_compose.py rpi -o device-compose-files/docker-compose-rpi.yml
-          cp device-compose-files/docker-compose-rpi.yml docker-compose.yml
-      - name: Generate Docker Compose for Pisces
-        if: matrix.sbc == 'pisces'
-        run: |
-          python gen_docker_compose.py pisces -o device-compose-files/docker-compose-pisces.yml
-          cp device-compose-files/docker-compose-pisces.yml docker-compose.yml
-      - name: Balena Deploy (Nebra ROCK Pi)
+          python gen_docker_compose.py ${{matrix.variant}} -o docker-compose.yml
+      - name: Balena Deploy Testnet
         uses: nebraltd/balena-cli-action@v13.3.0
-        if: matrix.sbc == 'rockpi' && success()
-        with:
           balena_api_token: ${{secrets.BALENA_API_TOKEN}}
-          balena_command: "deploy nebraltd/helium-testnet${{ env.ROCKPI }} --logs --debug --nocache --build"
-      - name: Balena Deploy (Nebra RasPi)
-        uses: nebraltd/balena-cli-action@v13.3.0
-        if: matrix.sbc == 'raspi' && success()
-        with:
-          balena_api_token: ${{secrets.BALENA_API_TOKEN_1}}
-          balena_command: "deploy nebraltd/helium-testnet --logs --debug --nocache --build"
-      - name: Balena Deploy (RAK and other non-Nebra RasPi)
-        uses: nebraltd/balena-cli-action@v13.3.0
-        if: success() && (matrix.sbc == 'rak' || matrix.sbc == 'og' || matrix.sbc == 'sensecap' || matrix.sbc == 'finestra' || matrix.sbc == 'controllino' || matrix.sbc == 'pisces' || matrix.sbc == 'cotx' || matrix.sbc == 'pantherx1')
-        with:
-          balena_api_token: ${{secrets.BALENA_API_TOKEN_2}}
-          balena_command: "deploy nebraltd/helium-testnet-${{ matrix.sbc }} --logs --debug --nocache --build"
-      - name: Push updated docker compose file (ROCK Pi)
```

Comment:
We need to keep these or it will break things as already discussed

Reply:
I'm changing things to create everything in one step. Will introduce it soon.